src/test/java/io/gravitee/policy/json2json/JsonToJsonTransformationPolicyIntegrationTest.java

```diff
@@ -15,203 +15,138 @@
-import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.when;
 
 import com.bazaarvoice.jolt.JsonUtils;
-import io.gravitee.common.http.MediaType;
-import io.gravitee.el.TemplateContext;
-import io.gravitee.el.TemplateEngine;
-import io.gravitee.gateway.api.ExecutionContext;
-import io.gravitee.gateway.api.Request;
-import io.gravitee.gateway.api.Response;
-import io.gravitee.gateway.api.buffer.Buffer;
-import io.gravitee.gateway.api.http.HttpHeaderNames;
-import io.gravitee.gateway.api.http.HttpHeaders;
-import io.gravitee.gateway.api.stream.ReadWriteStream;
-import io.gravitee.policy.api.PolicyChain;
+import io.gravitee.apim.gateway.tests.sdk.AbstractPolicyTest;
+import io.gravitee.apim.gateway.tests.sdk.annotations.DeployApi;
+import io.gravitee.apim.gateway.tests.sdk.annotations.GatewayTest;
+import io.gravitee.apim.gateway.tests.sdk.configuration.GatewayConfigurationBuilder;
 import io.gravitee.policy.json2json.configuration.JsonToJsonTransformationPolicyConfiguration;
-import io.gravitee.policy.json2json.configuration.PolicyScope;
-import io.gravitee.reporter.api.http.Metrics;
-import java.io.IOException;
+import io.vertx.rxjava3.core.buffer.Buffer;
+import io.vertx.rxjava3.core.http.HttpClient;
+import io.vertx.rxjava3.core.http.HttpClientRequest;
 import java.io.InputStream;
-import java.io.StringWriter;
-import java.time.Instant;
-import org.apache.commons.io.IOUtils;
-import org.junit.jupiter.api.BeforeEach;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
-import org.junit.jupiter.api.extension.ExtendWith;
-import org.mockito.Mock;
-import org.mockito.Spy;
-import org.mockito.junit.jupiter.MockitoExtension;
 
 /**
  * @author Yann TAVERNIER (yann.tavernier at graviteesource.com)
  * @author GraviteeSource Team
  */
-@ExtendWith(MockitoExtension.class)
-public class JsonToJsonTransformationPolicyIntegrationTest {
+@GatewayTest
+public class JsonToJsonTransformationPolicyIntegrationTest
+    extends AbstractPolicyTest<JsonToJsonTransformationPolicy, JsonToJsonTransformationPolicyConfiguration> {
 
-    @Mock
-    private JsonToJsonTransformationPolicyConfiguration jsonToJsonTransformationPolicyConfiguration;
-
-    private JsonToJsonTransformationPolicy jsonToJsonTransformationPolicy;
-
-    @Mock
-    protected ExecutionContext executionContext;
-
-    @Mock
-    private PolicyChain policyChain;
-
-    @Spy
-    private Request request;
-
-    @Spy
-    private Response response;
-
-    @BeforeEach
-    public void init() {
-        jsonToJsonTransformationPolicy = new JsonToJsonTransformationPolicy(jsonToJsonTransformationPolicyConfiguration);
-        when(executionContext.getTemplateEngine()).thenReturn(new MockTemplateEngine());
+    @Override
+    protected void configureGateway(GatewayConfigurationBuilder gatewayConfigurationBuilder) {
+        super.configureGateway(gatewayConfigurationBuilder);
```

Comment:
maybe I miss something but since you do not modify the configuration explicitly you don't need this

Reply:
Indeed, I was planning to activate jupiter mode but I will remove it